x-pack/plugins/infra/public/pages/metrics/hosts/components/kpis/kpi.tsx

```diff
@@ -57,19 +45,33 @@ export const Kpi = ({ id, title, layers, toolTip, height }: KPIChartProps & { he
     filters,
   });
 
+  const tooltipComponent = useMemo(() => <TooltipContent description={toolTip} />, [toolTip]);
+  const layerWithSubtitle = useMemo(() => {
+    const subtitle =
+      searchCriteria.limit < (hostCountData?.count.value ?? 0)
+        ? i18n.translate('xpack.infra.hostsViewPage.kpi.subtitle.average.limit', {
+            defaultMessage: 'Average (of {limit} hosts)',
+            values: {
+              limit: searchCriteria.limit,
+            },
+          })
+        : AVERAGE_SUBTITLE;
+    return { ...layers, options: { ...layers.options, subtitle } };
+  }, [hostCountData?.count.value, layers, searchCriteria.limit]);
```

Comment:
After replacing the `useLazyRef` with `useMemo` in the [use_lens_attributes.ts](https://github.com/elastic/kibana/pull/166276/files#diff-3b8924df93938d7c47bbc5b2af6eb3e7b96d23f300e6e6a84ac0fd747bbacffe). I had to change several places to make sure charts were not unnecessarily re-rendering.